### DISCUSSION TITLE
Is it possible to get pathname relative to basename ?

### DISCUSSION THREAD

**Discussion by p-l-l** [remix-run/react-router#8068] [Q&A] [answered]
I have an application (using react-router v6) in which routes are defined as enums in a simple way e.g. '/home'.
The routes are rendered in a Routes component with a basename value e.g. '/app'.

I would like to detect if my list item recieving the path '/home' matches the current location.
I used useLocation() to get the pathname but I had to replace basename from pathname the get it to match
with '/home' (else it would compare pathname='/app/home' to '/home' param of list item).

Is there a more elegant/built-in way to do this ?

Thanks for any help.

**Comment by mjackson** [comment, marked as answer]
This was fixed in v6.0.0-beta.5 (see [the release notes](https://github.com/remix-run/react-router/releases/tag/v6.0.0-beta.5)). The `basename` is now stripped from `location.pathname` in `useLocation()`.

**Reply by tisonkun** [reply]
If so, how can I get the original pathname now?
